pkg/kubelet/kuberuntime/kuberuntime_sandbox.go

```diff
@@ -138,6 +140,14 @@ func (m *kubeGenericRuntimeManager) generatePodSandboxConfig(pod *v1.Pod, attemp
 	}
 	podSandboxConfig.Linux = lc
 
+	if runtime.GOOS == "windows" {
```

Comment:
We want to set `CredentialSpec` and `RunAsUsername` even when the `WindowsHostProcessContainers` feature flag is not set.

For some time containerD will be looking for the fields under both LinuxOptions and WindowsOptions (preferring WindowsOptions if they exist). We're still working out the timelines for that tho...

Reply:
Actually this is more related to https://github.com/kubernetes/kubernetes/issues/92963 than #93580 but it will help with that one too. These CRI updates were discussed in the KEP https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1981-windows-privileged-container-support#cri-implementation-details